applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp

```diff
@@ -1542,43 +1543,35 @@ void UPwSmallStrainElement<TDim, TNumNodes>::CalculateExtrapolationMatrix(Bounde
 template <>
 void UPwSmallStrainElement<2, 3>::CalculateExtrapolationMatrix(BoundedMatrix<double, 3, 3>& rExtrapolationMatrix)
 {
-    // The matrix contains the shape functions at each GP evaluated at each
-    // node. Rows: nodes Columns: GP
-
     // Triangle_2d_3
     // GI_GAUSS_2
-    rExtrapolationMatrix(0, 0) = 1.6666666666666666666;
-    rExtrapolationMatrix(0, 1) = -0.33333333333333333333;
-    rExtrapolationMatrix(0, 2) = -0.33333333333333333333;
-    rExtrapolationMatrix(1, 0) = -0.33333333333333333333;
-    rExtrapolationMatrix(1, 1) = 1.6666666666666666666;
-    rExtrapolationMatrix(1, 2) = -0.33333333333333333333;
-    rExtrapolationMatrix(2, 0) = -0.33333333333333333333;
-    rExtrapolationMatrix(2, 1) = -0.33333333333333333333;
-    rExtrapolationMatrix(2, 2) = 1.6666666666666666666;
+    const auto extrapolator = LinearNodalExtrapolator{};
+    const auto result       = extrapolator.CalculateElementExtrapolationMatrix(
+        this->GetGeometry(), this->GetIntegrationMethod());
+    KRATOS_ERROR_IF_NOT(result.size1() == 3)
+        << "Extrapolation matrix has unexpected number of rows: " << result.size1()
+        << " (expected 3)" << std::endl;
+    KRATOS_ERROR_IF_NOT(result.size2() == 3)
+        << "Extrapolation matrix has unexpected number of columns: " << result.size1()
+        << " (expected 3)" << std::endl;
+    rExtrapolationMatrix = result;
 }
```

Comment:
The number of rows and colums are checked here. However, they are hard programmed to be allocated and filled correctly. So, I think checking a correctly allocated arrays is just extra work and time for the code and increases the line in the code. What do you think about it?

Reply:
I have added these checks, since the "extrapolator" returns a matrix with dynamic sizes. In other words, its sizes are unknown at compile time. However, the very next action is to assign this variable-sized matrix to a matrix with fixed sizes, which are known at compile time (i.e. `Matrix` vs. `bounded_matrix<...>`). To avoid that we attempt to assign a matrix that has mismatching sizes, we explicitly check the variable-sized matrix.